### ISSUE TITLE
Allow generator-star-spacing to have optional spaces

### ISSUE THREAD

**Issue by rwwagner90** [eslint/eslint#6777]
**ESLint Version:** 
2.13.1

**Rule to change:**
`generator-star-spacing`: I would like to add functionality where you can specify only `before` or `after` and the one you leave off will be optionally space or no space acceptable.

**Code I would like to be acceptable:**

```js
function* foo() {}

let bar = function*() {};
```

**What happens without change:**
One or the other will fail, as if you specify `'generator-star-spacing': ['error', {'before': false}],` after defaults to false, rather than being an optional space.

Essentially, it should function the same as this rule in JSCS:
```json
"disallowSpacesInGenerator": {
     "beforeStar": true
 },
```

**Comment by vitorbal** [comment]
Hi there @rwwagner90, thanks for reaching out!

Could you do us a big favor and fill in the [template for proposing a rule change](http://eslint.org/docs/developer-guide/contributing/rule-changes#proposing-a-rule-change)?
That will help the team understand what you are proposing and allow us to give quicker feedback on your proposal. Thanks a lot!

**Comment by rwwagner90** [comment]
Hi @vitorbal, I just tried to update to use the template, but my rule is the opposite of the template. I want to define what should pass, not be stricter on what should fail. Please let me know if my description is adequate.

**Comment by platinumazure** [comment]
@rwwagner90 Thanks for calling out the fact that the rule change template implies only stricter changes might be allowed. I've observed that independently and I do intend to raise the issue with the team. I think what you have in your post is reasonably clear. Thanks for bearing with us!

**Comment by rwwagner90** [comment]
No problem @platinumazure! Thanks for looking into this 😄

**Comment by rwwagner90** [comment]
Any updates on whether this will be supported or not? If not, we could probably write our own custom rules that handle this, and not use this rule.

**Comment by platinumazure** [comment]
@rwwagner90 We require a champion and three :+1: endorsements from ESLint team members in order to accept the proposal. I'll offer a :+1: to help that along.

**Comment by alberto** [comment]
Thanks for your interest in improving eslint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
